src/mutex.rs

```diff
@@ -324,13 +324,12 @@ impl<T: 'static + ?Sized> Mutex<T> {
     ///         *guard += 5;
     ///         Ok(()) as Result<(), ()>
     ///     })
-    ///     .map(|_| assert_eq!(mtx.try_unwrap().unwrap(), 5))
     /// }));
-    /// assert!(r.is_ok());
```

Comment:
Isn't this going to cause a warning about `r` being unused?  And if you remove `r`, then there'll be a warning about `current_thread::block_on_all` returning a `Result` that needs to be used?  You should probably restore the assertion.